bot/exts/moderation/stream.py

```diff
@@ -0,0 +1,139 @@
+import datetime
+
+import discord
+from async_rediscache import RedisCache
+from discord.ext import commands
+
+from bot.bot import Bot
+from bot.constants import Emojis, Guild, Roles, STAFF_ROLES, VideoPermission
+from bot.converters import Expiry
+from bot.utils.scheduling import Scheduler
+from bot.utils.time import format_infraction_with_duration
+
+
+class Stream(commands.Cog):
+    """Grant and revoke streaming permissions from users."""
+
+    # Stores tasks to remove streaming permission
+    # User id : timestamp relation
+    task_cache = RedisCache()
+
+    def __init__(self, bot: Bot):
+        self.bot = bot
+        self.scheduler = Scheduler(self.__class__.__name__)
+        self.reload_task = self.bot.loop.create_task(self._reload_tasks_from_redis())
+
+    async def _remove_streaming_permission(self, schedule_user: discord.Member) -> None:
+        """Remove streaming permission from Member."""
+        await self._delete_from_redis(schedule_user.id)
+        await schedule_user.remove_roles(discord.Object(Roles.video), reason="Streaming access revoked")
+
+    async def _add_to_redis_cache(self, user_id: int, timestamp: float) -> None:
+        """Adds 'task' to redis cache."""
+        await self.task_cache.set(user_id, timestamp)
+
+    async def _reload_tasks_from_redis(self) -> None:
+        await self.bot.wait_until_guild_available()
+        items = await self.task_cache.items()
+        for key, value in items:
+            member = await self.bot.get_guild(Guild.id).fetch_member(key)
```

Comment:
You should try to use `get_member` before resorting to using an API call.

Also, you should catch the exception this may raise. Separate the handling of the member not being found from other network issues. If the member isn't found, that can just be logged at debug level and ignored. If there is a different exception, then that should be logged with `log.exception`. Note that member not found has a code of 10007, which is not the same at `NotFound` / an HTTP 404 status.